scripts/build_local_binary.sh

```diff
@@ -0,0 +1,30 @@
+# When you run a `make bin` in the terraform-provider-rancher2 repo, it
+# downloads a binary into your local cache from the specified source. To
+# tell terraform to use a local binary, you need to overwrite the one in
+# the cache.
+
+# Add the following to your terraform config file. This tells terraform 
+# to use the local cache https://terraform.io/cli/config/config-file.
+
+```
+terraform {
+  required_providers {
+    rancher2 = {
+      source = "terraform.example.com/local/rancher2"
+      version = "1.0.0"
+    }
+  }
+}
+```
+
+# Compile the local binary
+make bin
+
+# Overwrite the cached binary with your local one
+cp terraform-provider-rancher2 ~/.terraform.d/plugins/terraform.example.com/local/rancher2/1.0.0/darwin_amd64/terraform-provider-rancher2
```

Comment:
Suggestion: make version and platform customizable, so people using Apple Silicon and/or Linux can also use it. Example: `${HOME}/.terraform.d/plugins/terraform.example.com/local/rancher2/${VERSION}/$(uname | tr '[:upper:]' '[:lower:]')_$(arch)"`

Reply:
Note: On amd64 `$(arch)` outputs `x86_64` instead of `amd64`. And on M1 Macs I think it will output `aarch64` instead of `arm64`. I'm not familiar with terraform plugins, the suggestion might work fine. But I know rancher uses `amd64` and `arm64` instead of `x86_64` and `aarch64`.

Reply:
Interesting! `uname -m` might be a good alternative. My M1 Mac outputs `arm64` for that, but it does output `aarch64` for Linux (slight difference from Darwin to Linux).

Reply:
It seems `uname -m` also outputs `x86_64` for `amd64` on Linux so I guess I have to update my suggestion to having a `case` statement instead 😞

Reply:
Yeah it's a mess! I don't think it's possible to use a single command, it might be possible to use a one liner with sed like, `$(arch | sed s/x86_64/amd64/ | sed s/aarch64/arm64/)` though.

Reply:
I'd vote on having a case statement in this situation so we can cover more scenarios in the future if needed be. Like the kustomize install script: https://github.com/kubernetes-sigs/kustomize/blob/78978723db943fe5fe4baebd043dd9ab02ab144d/hack/install_kustomize.sh#L108-L124